### PR TITLE
Support converting integral types in hex and octal

### LINKED ISSUE #1624: Support hexadecimal values in arguments converters
## Overview

It’s a feature request: support converting argument strings containing integral literals in hexadecimal to the corresponding types on top of decimal.

### Description
Currently one cannot use hexadecimal integral values when specifying arguments, e.g., this won't work:
```java
@CsvSource({
    "1",
    "0x1", // Won't be converted
}) // CSV source for illustrative purposes: ValueSource would suite here better
@ParameterizedTest
void test(int v) {
```

The cases when one needs to use hex include copying values in hex from some specification; when binary representation (i.e., which bit in the value is set) matters.

### Workaround 
Use String as argument type and convert yourself with, e.g., Integer::decode.

## Deliverables

- [ ] ...

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Use PrimitiveType#decode to convert strings to the primitive
integral types. This allows the values to be specified in the following
format:
```
    DecodableString:
        Signopt DecimalNumeral
        Signopt 0x HexDigits
        Signopt 0X HexDigits
        Signopt # HexDigits
        Signopt 0 OctalDigits
    Sign:
        -
        +
```

Addresses #1624

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
Updated. I am not sure where to mention various formats of integers — probably putting each accepted format in each cell corresponding to an integral type would be too much (and I don't know if that's possible with this markup language).

File: documentation/src/docs/asciidoc/user-guide/writing-tests.adoc [hunk 1]
```diff
@@ -1061,6 +1061,7 @@ include::{testDir}/example/ParameterizedTestDemo.java[tags=implicit_conversion_e
 ----
 
 `String` instances are currently implicitly converted to the following target types.
+Integral types support decimal, hexadecimal and octal `String` literals.
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/user-guide/writing-tests.adoc]: I think we should add the concrete types in parenthesis like this:

```asciidoctor
Integral types (`byte`, `short`, `int`, `long`) support ...
```

File: documentation/src/docs/asciidoc/release-notes/release-notes-5.4.0-M1.adoc [hunk 2]
```diff
@@ -47,7 +47,8 @@ repository on GitHub.
 
 ==== New Features and Improvements
 
-* ❓
+* Support converting integral types in hex and octal in parameterized test
+  arguments resolvers.
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/release-notes/release-notes-5.4.0-M1.adoc]: Please add 

```
(e.g. `"0xff"` into `255`)
```

File: documentation/src/docs/asciidoc/user-guide/writing-tests.adoc [hunk 3]
```diff
@@ -1070,7 +1071,7 @@ include::{testDir}/example/ParameterizedTestDemo.java[tags=implicit_conversion_e
 | `boolean`/`Boolean`        | `"true"`                                 -> `true`
 | `byte`/`Byte`              | `"1"`                                    -> `(byte) 1`
 | `char`/`Character`         | `"o"`                                    -> `'o'`
-| `short`/`Short`            | `"1"`                                    -> `(short) 1`
+| `short`/`Short`            | `"0xF"`                                  -> `(short) 15`
```
[the remark this answers is not in the document]
Comment (dmitry-timofeev) [reply documentation/src/docs/asciidoc/user-guide/writing-tests.adoc]: Thanks, I like this format!

**Comment by marcphilipp** [comment]
Thanks, @dmitry-timofeev! 👍

**Pull request merged**